library/net.cpp

```diff
@@ -1100,6 +1100,8 @@ static int dnet_socket_connect(dnet_node *node, dnet_addr_socket_set &original_l
 	}
 
 	pthread_mutex_lock(&state->lock);
+
+	assert(state->succeed_count + state->failed_count <= state->total_count); // 'less' could be on timeout
```

Comment:
We always build with debug support. Maybe we do not set special compile-time definitions, but in this case it is useless - we do not make separate builds for tests, don't we?